sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -799,37 +799,40 @@ private[lf] final class Compiler(
       soft: Boolean,
   ): (t.SDefinitionRef, SDefinition) =
     topLevelFunction2(t.FetchInterfaceDefRef(ifaceId)) { (cidPos, _, env) =>
-      let(
-        env,
-        (if (soft) SBSoftFetchInterface else SBFetchAny(None))(
-          env.toSEVar(cidPos),
-          s.SEValue.None,
-        ),
-      ) { (payloadPos, env) =>
+      if (soft) {
+        SBFetchInterface(ifaceId)(env.toSEVar(cidPos))
```

Comment:
I'll rename it to something that contains the word "Soft" in the PR that updates exercise by interface, and I'll share the code between the two.